geode-core/src/main/java/org/apache/geode/internal/cache/RegionPerfStats.java

```diff
@@ -14,27 +14,55 @@
  */
 package org.apache.geode.internal.cache;
 
+import java.util.concurrent.atomic.AtomicLong;
 import java.util.function.LongSupplier;
 
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+
 import org.apache.geode.StatisticsFactory;
 import org.apache.geode.annotations.VisibleForTesting;
+import org.apache.geode.cache.DataPolicy;
 import org.apache.geode.internal.NanoTimer;
 
 class RegionPerfStats extends CachePerfStats {
 
   private final CachePerfStats cachePerfStats;
+  private final MeterRegistry meterRegistry;
+  private final Gauge entriesGauge;
+  private final AtomicLong entryCount;
```

Comment:
Converting from a `LongAdder` to `AtomicLong` will result in poor entry update performance. Unless you need to set specific values you should be using `LongAdder` for multithreaded increments and decrements.

Reply:
Done